langchain4j-bedrock/src/main/java/dev/langchain4j/model/bedrock/BedrockTitanEmbeddingModel.java

```diff
@@ -26,12 +37,40 @@ protected String getModelId() {
         return model;
     }
 
+    /**
+     * 1024 is default size of output vector for Titan Embedding model V2.
+     */
+    @Builder.Default
+    private final Integer dimensions = 1024;
+
+    /**
+     *  A flag indicating whether or not to normalize the output embeddings.
+     *  It defaults to true, which is optimal for RAG use cases.
+     */
+    @Builder.Default
+    private final Boolean normalize = true;
+
+
     @Override
     protected List<Map<String, Object>> getRequestParameters(List<TextSegment> textSegments) {
-        return textSegments.stream()
-                .map(TextSegment::text)
-                .map(text -> of("inputText", text))
-                .collect(Collectors.toList());
+        if (MODEL_V1_ID.equals(this.model)) {
+            return textSegments.stream()
+                    .map(TextSegment::text)
+                    .map(text -> of("inputText", text))
+                    .collect(Collectors.toList());
+        } else if (MODEL_V2_ID.equals(this.model)) {
```

Comment:
Is `MODEL_V2_ID` the only model that supports dimensions/normalize?
Would be nice to think about how to make it future-proof when new models will be added (they will most probably support these params).

BTW, if dimensions/normalize are not supported by the older model, it would be better to fail (either validate on our side or send them to AWS so that it can validate) in this case, otherwise user might be expecting for parameters to work, but they will be silently ignored

Reply:
I just checked ... if we provide `dimensions` and `normalize` as parameters then AWS Bedrock throws 400 Exception.
example.
`Malformed input request: 2 schema violations found, please reformat your input and try again. (Service: BedrockRuntime, Status Code: 400, Request ID: b606929b-e642-4c34-8565-bc4e32f66510)
`
But I agree to all your comments.
I removed IllegalArgumentException and now just checking if V1 model is applied.